### PR TITLE
Week four 3

### PR DESCRIPTION
**Pull request by Sarahhud**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/BlobstoreUploadUrlServlet.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+@WebServlet("/blobstore-upload-url")
+public class BlobstoreUploadUrlServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    String uploadUrl = blobstoreService.createUploadUrl("/new-comment");
+
+    response.setContentType("text/html");
```
[the remark this answers is not in the document]
Comment (Sarahhud) [reply portfolio/src/main/java/com/google/sps/servlets/BlobstoreUploadUrlServlet.java:37]: So Nic and I tried using Json but it just keeps messing up the url it's returning when we try to convert the json back into a string. We couldn't figure out what's causing the problem, but we did try! I also noticed that in all the examples they are using text/html to return urls not json...

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -48,8 +48,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       long timestamp = (long) entity.getProperty("timestamp");
       String text = (String) entity.getProperty("text");
       String email = (String) entity.getProperty("email");
-
-      Comment comment = new Comment(id, title, timestamp, text, email);
+      String url = null;
+      if(entity.getProperty("image") != null){
```
Comment (NicholasCobbFromGoogle) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: nit: "if (entity[...]"

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -97,3 +107,12 @@ function deleteComment(comment) {
   params.append('id', comment.id);
   fetch('/delete-comment', {method: 'POST', body: params});
 }
+
+async function fetchBlobstoreUrlAndShowForm() {
+  fetch('/blobstore-upload-url').then(response => response.text())
+     .then((imageUploadUrl) => {
+     const messageForm = document.getElementById('my-form');
+     messageForm.action = String(imageUploadUrl);
+     messageForm.classList.remove('hidden');
```
Comment (NicholasCobbFromGoogle) [inline portfolio/src/main/webapp/script.js:116]: nit: I think the indentation needs to be fixed here.

Comment (Sarahhud) [reply portfolio/src/main/webapp/script.js]: Like that? @NicholasCobbFromGoogle

Comment (NicholasCobbFromGoogle) [reply portfolio/src/main/webapp/script.js]: Yep LGTM

**Pull request merged**